src/filholes.jl

```diff
@@ -0,0 +1,58 @@
+"""
+    fillhole(img; [dims])
+    fillhole(img; se)
+
+Fill the holes in image 'img'. Could be binary or grascale
+
+The `dims` keyword is used to specify the dimension to process by constructing the box shape
+structuring element [`strel_box(img; dims)`](@ref strel_box). For generic structuring
+element, the half-size is expected to be either `0` or `1` along each dimension.
+
+The output has the same type as input image
+"""
+
+function fillhole(img; dims=coords_spatial(img))
+    return fillhole(img, strel_box(img, dims))
+end
+
+function fillhole(img, se)
+    return fillhole!(similar(img), img, se)
+end
+
+function fillhole!(out, img; dims=coords_spatial(img))
+    return fillhole!(out, img, strel_box(img, dims))
+end
+
+function fillhole!(out, img, se)
+    return _fillhole!(out, img, se)
```

Comment:
Why do we need the underscore prefix? Could we combine the function below and this one to a single signature?

Reply:
The prefix correspon to the most internal call, you will see this pattern in the whole project, something like "private" implem
But yes you're right in this special case its not relevant